client/js/widget-manager.js

```diff
@@ -24,17 +24,21 @@ var Widgets = require('jupyter-js-widgets');
         // Register the comm target
         this.commManager.register_target(this.comm_target_name, this.handle_comm_open.bind(this));
 
-        // Validate the version requested by the backend.
-        var validate = (function validate() {
-            this.validateVersion().then(function(valid) {
-                if (!valid) {
-                    console.warn('Widget frontend version does not match the backend.');
-                }
-            }).catch(function(err) {
-                console.error('Could not cross validate the widget frontend and backend versions.', err);
-            });
-        }).bind(this);
-        validate();
+        // Validate the version requested by the backend -- required for ipywidgets
+        // If kernel name is not defined, default to python3
+        kernelname = kernelname.toLowerCase();
+        if (!kernelname || kernelname === 'python3' || kernelname === 'python2') {
```

Comment:
@jhpedemonte also clarified that the validation check does not actually inhibit the Toree kernel from working. It's just an ugly error. We agreed that switching it to a warning that explicitly says ala "Compatible version of ipywidgets is not available" is cleaner than hardcoding python checks and still makes sense with other kernels (yes, ipywidgets is not available for R, Toree, etc.)